Shared/ViewModels/AdminDashboard/ActiveSessionsViewModel.swift

```diff
@@ -13,105 +13,57 @@ import OrderedCollections
         case refresh
         case backgroundRefresh
-    }
 
-    // MARK: - BackgroundState
+        var transition: Transition {
+            .loop(.refreshing)
+        }
+    }
 
     enum BackgroundState: Hashable {
         case backgroundRefreshing
     }
 
-    // MARK: - State
-
-    enum State: Hashable {
-        case content
-        case error(JellyfinAPIError)
+    enum State {
         case initial
+        case error
+        case refreshing
     }
 
     @Published
     var activeWithinSeconds: Int? = 900 {
         didSet {
-            send(.refresh)
+            refresh()
         }
     }
 
     @Published
     var showSessionType: ActiveSessionFilter = .all {
         didSet {
-            send(.refresh)
+            refresh()
         }
     }
 
-    @Published
-    var backgroundStates: Set<BackgroundState> = []
     @Published
     var sessions: OrderedDictionary<String, BindingBox<SessionInfoDto?>> = [:]
-    @Published
-    var state: State = .initial
-
-    private var sessionTask: AnyCancellable?
-
-    func respond(to action: Action) -> State {
-        switch action {
-        case .backgroundRefresh:
-            sessionTask?.cancel()
-
-            sessionTask = Task { [weak self] in
-                await MainActor.run {
-                    let _ = self?.backgroundStates.insert(.backgroundRefreshing)
-                }
-
-                do {
-                    try await self?.updateSessions()
-                } catch {
-                    guard let self else { return }
-                    await MainActor.run {
-                        self.state = .error(.init(error.localizedDescription))
-                    }
-                }
-
-                await MainActor.run {
-                    let _ = self?.backgroundStates.remove(.backgroundRefreshing)
-                }
-            }
-            .asAnyCancellable()
-
-            return state
-        case .refresh:
-            sessionTask?.cancel()
-
-            sessionTask = Task { [weak self] in
-                await MainActor.run {
-                    self?.state = .initial
-                }
-
-                do {
-                    try await self?.updateSessions()
-
-                    guard let self else { return }
-
-                    await MainActor.run {
-                        self.state = .content
-                    }
-                } catch {
-                    guard let self else { return }
-                    await MainActor.run {
-                        self.state = .error(.init(error.localizedDescription))
-                    }
-                }
-            }
-            .asAnyCancellable()
 
-            return .initial
-        }
+    @Function(\Action.Cases.refresh)
+    private func _refresh() async throws {
+        try await updateSessions()
+    }
+
+    @Function(\Action.Cases.backgroundRefresh)
+    private func _backgroundRefresh() async throws {
+        backgroundStates.insert(.backgroundRefreshing)
```

Comment:
For background states, these are specified in the `Action.transition` value. The state system will automatically insert and remove the background state as necessary.

```swift
var transition: Transition {
    switch self {
    case .refresh: .loop(.refreshing)
    case .backgroundRefresh: .background(.backgroundRefreshing)
    }
}
```